source/code/plugin/filter_inventory2mdm.rb

```diff
@@ -0,0 +1,261 @@
+# Copyright (c) Microsoft Corporation.  All rights reserved.
+
+# frozen_string_literal: true
+
+module Fluent
+    require 'logger'
+    require 'json'
+    require_relative 'oms_common'
+
+	class Inventory2MdmFilter < Filter
+		Fluent::Plugin.register_filter('filter_inventory2mdm', self)
+		
+		config_param :enable_log, :integer, :default => 0
+        config_param :log_path, :string, :default => '/var/opt/microsoft/docker-cimprov/log/filter_inventory2mdm.log'
+        config_param :custom_metrics_azure_regions, :string
+        
+        @@node_count_metric_name = 'nodesCount'
+        @@pod_count_metric_name = 'podCount'
+        @@pod_inventory_tag = 'mdm.kubepodinventory'
+        @@node_inventory_tag = 'mdm.kubenodeinventory'
+        @@node_status_ready = 'Ready'
+        @@node_status_not_ready = 'NotReady'
+
+        @@node_inventory_custom_metrics_template = '
+            { 
+                "time": "%{timestamp}", 
+                "data": { 
+                    "baseData": { 
+                        "metric": "%{metricName}", 
+                        "namespace": "insights.container/nodes", 
+                        "dimNames": [ 
+                        "status"
+                        ], 
+                        "series": [ 
+                        { 
+                            "dimValues": [ 
+                            "%{statusValue}"
+                            ], 
+                            "min": %{node_status_count},
+                            "max": %{node_status_count}, 
+                            "sum": %{node_status_count}, 
+                            "count": 1
+                        } 
+                        ] 
+                    } 
+                } 
+            }'
+
+        @@pod_inventory_custom_metrics_template = '
+            { 
+                "time": "%{timestamp}", 
+                "data": { 
+                    "baseData": { 
+                        "metric": "%{metricName}", 
+                        "namespace": "insights.container/nodes", 
```

Comment:
can this go to 'pods' namespace instead of 'nodes' ?

Reply:
that will create too many namespaces. I think it makes sense to have one single namespace for container insights. This will also cause discoverability issues. We can't expect customers to go check every namespace to see what metrics each one has. Just my thoughts

Reply:
#1 ) There are not much restrictions on the num namespaces
#2) So you expect customers to look for podcount under /nodes (and will make it easier to discover than putting them under 'pods' ? In the futuire when we add pod cpu, we will add it under nodes? then how about node cpu ? If i am looking for charting anything podlevel its easy to understand to go to pods namespace.
3) either flatten everything and put under default namespace or catagorize it properly.